frontend/src/components/object/ObjectFilters.vue

```diff
@@ -0,0 +1,138 @@
+<script setup lang="ts">
+import { storeToRefs } from 'pinia';
+import { ref, computed } from 'vue';
+
+import { MultiSelect } from '@/lib/primevue';
+import { useMetadataStore, useObjectStore, useTagStore } from '@/store';
+
+import type { MetadataPair, Tag } from '@/types';
+
+// Props
+type Props = {
+  bucketId?: string;
+};
+
+const props = withDefaults(defineProps<Props>(), {
+  bucketId: undefined,
+});
+
+// Store
+const metadataStore = useMetadataStore();
+const objectStore = useObjectStore();
+const tagStore = useTagStore();
+const { getMetadataSearchResults } = storeToRefs(useMetadataStore());
+const { getTagSearchResults } = storeToRefs(useTagStore());
+
+// State
+const searching = ref(false);
+const selectedMetadata = ref([]);
+const selectedTags = ref([]);
+
+// Computed
+const metadataValues = computed(() => {
+  // Take the metadata for all objects, distinct, and flatten them into a single array
+  // Add a display property to each tag to be used by the multiselect
+  const distinctVals = [
+    ...new Set(getMetadataSearchResults.value.flatMap((obj) => obj.metadata)),
+  ];
+  return distinctVals
+    .map((val) => ({ ...val, display: `${val.key}:${val.value}` }))
+    .filter(
+      (val, index, self) =>
+        index === self.findIndex((t) => t.display === val.display)
+    )
+    .sort((a, b) => a.display.localeCompare(b.display));
+});
+
+const tagsetValues = computed(() => {
+  // Take the tags for all objects, distinct, and flatten them into a single array
+  // Add a display property to each tag to be used by the multiselect
+  // coms-id not allowed as a tag to filter on by COMS
+  const distinctVals = [
+    ...new Set(getTagSearchResults.value.flatMap((obj) => obj.tagset)),
+  ];
+  return distinctVals
+    .map((val) => ({ ...val, display: `${val.key}=${val.value}` }))
+    .filter((val) => val.key !== 'coms-id')
+    .filter(
+      (val, index, self) =>
+        index === self.findIndex((t) => t.display === val.display)
+    )
+    .sort((a, b) => a.display.localeCompare(b.display));
+});
+
+// Actions
+const selectedFilterValuesChanged = () => {
+  // Get the 'display' property out from selected tag and metadata
+  const metaToSearch: Array<MetadataPair> = selectedMetadata.value.map(
+    ({ ...meta }: any) => meta
+  );
+  const tagSetToSearch: Array<Tag> = selectedTags.value.map(
+    ({ ...tag }: any) => tag
+  );
+
+  // Search the object store with the tagset as a param and metadata as headers
+  objectStore.fetchObjects(
+    {
+      bucketId: props.bucketId,
+    },
+    tagSetToSearch,
+    metaToSearch
+  );
+};
+
+const searchMetadata = async () => {
+  searching.value = true;
+  //await 4 seconds
+  await metadataStore.searchMetadata();
+  searching.value = false;
+};
+
+const searchTagging = async () => {
+  searching.value = true;
+  //await 4 seconds
+  await tagStore.searchTagging();
```

Comment:
I noticed that the dropdown includes tags for all objects, from all buckets in the system (on objects I have READ perm for).

UX question but I suspect we need to filter results of searchTagging to only show tags that are on objects in the current bucket.

You can pass an array of objectId's to the searchTagging endpoint but i dont think that's feasible (too many).
i suspect we need a new bucketId parameter on the COMS `/object/tagging `endpoint. 
but for now maybe we can filter the result of searchTagging against the list of objects in the objectStore (which are already filtered to current bucket, or have a bucketId property). I think applying the tag/metadata filters will filters the list of objects on the objectStore.. so the available tags in the dropdown will also be filtered.. which might be a good user experience.

god this is complicated.. i may have misunderstood all this.. and probably arent making much sense.

Reply:
I agree that we need to add `bucketId` scoping capability to that `/object/tagging` and `/object/metadata` endpoint so that we only get results relevant to this specific bucket, given the context of the page we're doing this on.

Reply:
Lets do like @TimCsaky says for now in here (just added code to filter) where we just filter out items from the metadata and tagging search results to exclude objectIDs that are not in the current list. This takes care of the deleted item issue as well.

I am of the opinion as well that the COMS search endpoints should probably provide bucket (and deleted) filters at the API level. But for now it's simple for BCBox as a consumer at least since (as Tim mentioned) the list of objects is already loaded there.